src/renderer/domains/network/block/store.ts

```diff
@@ -0,0 +1,76 @@
+import { type ApiPromise } from '@polkadot/api';
+import { attach, createEffect, createEvent, createStore, sample } from 'effector';
+import { interval } from 'patronum';
+
+import { type HexString } from '@/shared/core';
+import { getCurrentBlockNumber } from '@/shared/lib/utils';
+import { type BlockHeight, pjsSchema } from '@/shared/polkadotjs-schemas';
+import { networkModel } from '@/entities/network';
+
+export const $apis = networkModel.$apis;
+
+// Store mapping chain IDs to their current block heights
+const $currentBlock = createStore<Record<HexString, BlockHeight>>({});
+
+const startBlockListening = createEvent();
+const stopBlockListening = createEvent();
+
+// Create an effect factory for getting a block for a specific chain
+const getBlockForChainFx = attach({
+  effect: createEffect(async ({ chainId, api }: { chainId: HexString; api: ApiPromise }) => {
+    try {
+      const blockNumber = await getCurrentBlockNumber(api);
+      return {
+        chainId,
+        blockHeight: pjsSchema.helpers.toBlockHeight(blockNumber),
+      };
+    } catch (error) {
+      console.error(`Failed to get block for chain ${chainId}:`, error);
+      throw error;
```

Comment:
Currently there is no error handler `getBlockForChainFx.fail`, can safely omit `throw error`